front/lib/api/assistant/conversation.ts

```diff
@@ -1998,22 +1988,24 @@ async function isMessagesLimitReached({
   plan: PlanType;
   subscription: SubscriptionType;
 }): Promise<boolean> {
+  const { maxMessages, maxMessagesTimeframe } = plan.limits.assistant;
+
   if (plan.limits.assistant.maxMessages === -1) {
```

Comment:
This check was hindering the application of any limit because `maxMessages` was not overriden in the trial revision.